multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/mmrcms/MMRCMS.kt

```diff
@@ -1,529 +1,476 @@
-        val newPathSegments = parsedNewUrl.pathSegments.toMutableList()
+    override fun searchMangaSelector() = "div.media"
 
-        for (i in parsedBaseUrl.pathSegments) {
-            if (i.trim().equals(newPathSegments.first(), true)) {
-                newPathSegments.removeAt(0)
-            } else {
-                break
-            }
-        }
+    override fun searchMangaFromElement(element: Element) = SManga.create().apply {
+        val anchor = element.selectFirst(".media-heading a, .manga-heading a")!!
 
-        val builtUrl = parsedNewUrl.buildUpon().path("/")
-        newPathSegments.forEach { builtUrl.appendPath(it) }
+        setUrlWithoutDomain(anchor.attr("href"))
+        title = anchor.text()
+        thumbnail_url = MMRCMSUtils.guessCover(baseUrl, url, element.selectFirst("img")?.imgAttr())
+    }
 
-        var out = builtUrl.build().encodedPath!!
-        if (parsedNewUrl.encodedQuery != null) {
-            out += "?" + parsedNewUrl.encodedQuery
-        }
-        if (parsedNewUrl.encodedFragment != null) {
-            out += "#" + parsedNewUrl.encodedFragment
+    override fun searchMangaNextPageSelector(): String? = ".pagination a[rel=next]"
+
+    protected fun parseSearchDirectory(page: Int): MangasPage {
+        val manga = mutableListOf<SManga>()
+        val endRange = ((page * 24) - 1).let { if (it <= searchDirectory.lastIndex) it else searchDirectory.lastIndex }
+
+        for (i in (((page - 1) * 24)..endRange)) {
+            manga.add(
+                SManga.create().apply {
+                    url = "/$itemPath/${searchDirectory[i].data}"
+                    title = searchDirectory[i].value
+                    thumbnail_url = MMRCMSUtils.guessCover(baseUrl, url, null)
+                },
+            )
         }
 
-        return out
+        return MangasPage(manga, endRange < searchDirectory.lastIndex)
     }
 
+    protected open val detailAuthor = hashSetOf("author(s)", "autor(es)", "auteur(s)", "著作", "yazar(lar)", "mangaka(lar)", "pengarang/penulis", "pengarang", "penulis", "autor", "المؤلف", "перевод", "autor/autorzy")
+    protected open val detailArtist = hashSetOf("artist(s)", "artiste(s)", "sanatçi(lar)", "artista(s)", "artist(s)/ilustrator", "الرسام", "seniman", "rysownik/rysownicy", "artista")
+    protected open val detailGenre = hashSetOf("categories", "categorías", "catégories", "ジャンル", "kategoriler", "categorias", "kategorie", "التصنيفات", "жанр", "kategori", "tagi", "género")
+    protected open val detailStatus = hashSetOf("status", "statut", "estado", "状態", "durum", "الحالة", "статус")
+    protected open val detailStatusComplete = hashSetOf("complete", "مكتملة", "complet", "completo", "zakończone", "concluído", "finalizado")
+    protected open val detailStatusOngoing = hashSetOf("ongoing", "مستمرة", "en cours", "em lançamento", "prace w toku", "ativo", "em andamento", "activo")
+    protected open val detailStatusDropped = hashSetOf("dropped")
```

Comment:
Any reason why these are `open`?

Reply:
I copied Madara. `protected` because it's also used by Onma but there's probably no point to it being `open`.